tests/test_ble_state_manager.py

```
@@ -0,0 +1,671 @@
+    expected_operations = 5 * 100 * 3  # 5 workers * 100 iterations * 3 operations
+    assert (
+        total_operations >= expected_operations * 0.8
+    ), f"Too many failed operations: {total_operations}/{expected_operations}"
+
+    print(f"Contention performance: {total_operations} operations in {total_time:.3f}s")
+
+
+def test_memory_efficiency(self):
+    """Verify that BLEStateManager does not leak memory during creation and destruction."""
+    # Force garbage collection
+    gc.collect()
```

Ruff (0.14.1)

Undefined name `gc`

(F821)